src/main/java/ui/TestController.java

```diff
@@ -0,0 +1,74 @@
+package ui;
+
+import javafx.application.Application;
+
+import java.util.HashMap;
+
+public final class TestController {
+
+    private static UI ui;
+    private static HashMap<String, String> commandLineArgs;
+
+    private TestController() {}
+
+    public static void setUI(UI ui, Application.Parameters params) {
+        TestController.ui = ui;
+        TestController.commandLineArgs = initialiseCommandLineArguments(params);
+    }
+
+    private static HashMap<String, String> initialiseCommandLineArguments(Application.Parameters params) {
+        return new HashMap<>(params.getNamed());
+    }
+
+    public static boolean hasUI() {
+        return TestController.ui != null;
+    }
+
+    public static UI getUI() {
+        return ui;
+    }
+
+    public static HashMap<String, String> getCommandLineArgs() {
+        return commandLineArgs;
+    }
+
+    // Test mode should only be run as a test task (Gradle / JUnit), as quit()
+    // leaves the JVM alive during test mode (which is cleaned up by Gradle).
+    // Manually feeding --test=true into the command line arguments will leave the JVM
+    // running after the HT window has been closed, and thus will require the
+    // process to be closed manually afterwards (Force Quit / End Process).
```

Comment:
It's supposed to apply to not just the `isTestMode` method, but also the other subsequent command line argument methods.